### PR TITLE
releases: add TiDB 4.0.1 release notes

### PR DESCRIPTION
**Pull request by TomShawn**
### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

**If you select two or more versions from above**, to trigger the bot to cherry-pick this PR to your desired release version branch(es), you **must** add corresponding labels such as **needs-cherry-pick-4.0**, **needs-cherry-pick-3.1**, **needs-cherry-pick-3.0**, and **needs-cherry-pick-2.1**.

### What is the related PR or file link(s)?

- This PR is translated from: https://github.com/pingcap/docs-cn/pull/3600
- Other reference link(s): https://github.com/pingcap/tidb-prm/pull/14

### CODE REVIEW

File: releases/release-4.0.1.md [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+## Bug Fixes
+
++ TiKV
+
+    - Fix the issue that the `use-unified-pool` configuration in the startup log is incorrectly printed [#7946](https://github.com/tikv/tikv/pull/7946)
+    - Fix the issue that the tikv-ctl does not support relative path [#7963](https://github.com/tikv/tikv/pull/7963)
+    - Fix the bug that the monitoring metric of Point Selects is inaccurate [#8033](https://github.com/tikv/tikv/pull/8033)
+    - Fix the issue that a peer might not be destroyed after the network isolation disappears [#8006](https://github.com/tikv/tikv/pull/8006)
+    - Fix the issue that a `read index` request might be outdated [#8043](https://github.com/tikv/tikv/pull/8043)
```
Comment (overvenus) [inline releases/release-4.0.1.md:43]: The translation is wrong. It should be "Fix the issue that a read index request may get out-of-date commit index."

https://github.com/tikv/tikv/pull/8043

Comment (TomShawn) [reply releases/release-4.0.1.md:43]: @overvenus PTAL
```suggestion
    - Fix the issue that a read index request might get outdated commit index [#8043](https://github.com/tikv/tikv/pull/8043)
```

**Comment by TomShawn** [comment]
@overvenus PTAL again, thanks!

**Comment by scsldb** [comment]
LGTM

**Pull request merged**